src/lightning/data/streaming/dataset.py

```diff
@@ -68,12 +73,24 @@ def __init__(
 
         self.item_loader = item_loader
         self.shuffle: bool = shuffle
-        self.drop_last = drop_last
+        self.distributed_env = _DistributedEnv.detect()
+
+        if self.distributed_env.world_size > 1:
+            if drop_last is False:
+                logger.warn(
+                    "You're operating within a distributed environment and have enabled the `drop_last` option. "
```

Comment:
Looks flipped again, now it says 'you have enable the `drop_last` option' when it is actually false

Reply:
ChatGPT reverted it back ahah I should have verified instead of trusting it ;)

Reply:
done !